### ISSUE TITLE
[Feature request] Customization of tooltip values with data unavailable in Graphic's tuples

### ISSUE THREAD

**Issue by canastro** [entronad/graphic#85]
### TLDR
Allow the user to match a selected tuple with their own data-structures to more easily customize contents of the tooltip. 

### Detailed overview
In our app we are using a `tooltipRenderer` but we need to be able to customize the display format of each field based on information we don't have available in the tuples that are provided to Graphic. Therefore we need to get back the "index" of the selected tuple so that I can get the appropriate record from our data structure. Thats the gist of what we need. 

To give a bit more context, we're dynamically creating charts in a spreadsheet application with data provided by the user and, as google sheets we want to allow the user to have values with different formatting in each point. 

Here is an example where the user defined different number of decimal places per cell via formatting, ie. the user did not introduce 1.000000, the actual value is "1" but the formatting is "use 6 decimal places => 1.000000":

https://user-images.githubusercontent.com/1267623/163705323-2cd0d570-ce5c-448b-aeba-9c06f8d7c8ac.mov

### Possible hacky solution
Currently we could add an extra field to our "data" with the index of the current entry, but that seems hacky.
